src/aegis_ai/features/__init__.py

```diff
@@ -33,7 +36,32 @@ def __init__(self, agent: Agent):
     async def _run(self, call_str, prompt, **kwargs):
         try:
             runner = self.agent.run(prompt.to_string(), **kwargs)
-            return await asyncio.wait_for(runner, timeout=llm_prompt_timeout)
+
+            # Periodic warning logger while the run is in progress
+            done_event = asyncio.Event()
+
+            async def _warn_loop():
+                loop = asyncio.get_running_loop()
+                start_ts = loop.time()
+                while not done_event.is_set():
+                    try:
+                        await asyncio.wait_for(
+                            done_event.wait(), timeout=PROMPT_INFO_PERIOD
+                        )
+                    except asyncio.TimeoutError:
+                        elapsed = int(loop.time() - start_ts)
+                        logger.info(f"{call_str}: still running after {elapsed}s")
+
+            warn_task = asyncio.create_task(_warn_loop())
+            try:
+                return await asyncio.wait_for(runner, timeout=llm_prompt_timeout)
+            finally:
+                done_event.set()
+                warn_task.cancel()
+                try:
+                    await warn_task
+                except asyncio.CancelledError:
+                    pass
```

Comment:
The nesting here makes the code a little difficult to reason about, though I'm not sure refactoring into methods and passing parameters would be an improvement